database.md

```diff
@@ -0,0 +1,89 @@
+# Transcoder Database Schema
+
+Note that foreign keys constraints are not enforced at runtime, except in some tests.
+
+Tables:
+* kv
+* jobs
+* claims
+* receipts
+
+## Table `kv`
+
+Generic key-value table for miscellaneous data.
+
+Column | Type | Description
+--- | --- | ---
+key | STRING PRIMARY KEY |
+value | STRING |
+updatedAt | STRING DEFAULT CURRENT_TIMESTAMP |
+
+Values that may be set in the `kv` table:
+
+Key | Description
+--- | ---
+dbVersion |  The version of this database schema. Used to check compatibility and run migrations if needed.
+lastBlock | The last seen block.
+
+
+## Table `jobs`
+
+Stores information about jobs that the transcoder is assigned to.
+
+Column | Type | Description
+--- | --- | ---
+id |  INTEGER PRIMARY KEY| The ID of the on-chain job
+recordedAt | STRING DEFAULT CURRENT_TIMESTAMP | Time this row was inserted, *after* the job transaction has been confirmed on chain.
+streamID | STRING | Stream identifier, composed of `<NodeID><VideoID><Rendition>`.
+segmentPrice | INTEGER | The cost of transcoding each segment, in wei.
+transcodeOptions | STRING | [Encoded version of the rendition strings](https://github.com/livepeer/go-livepeer/blob/fac0393950e821f16555a17eb460b8abe74f6ada/common/videoprofile_ids.go#L5-L15).
+broadcaster | STRING | Broadaster's Eth address.
+transcoder | STRING | Transcoder's Eth address. Should be your node's address.
+startBlock | INTEGER | Block height from when the job was submitted on-chain.
+endBlock | INTEGER | Block height at which the job expires.
+stopReason | STRING | Reason for stopping a job on the transcoder. This may be before `endBlock` is reached. **Deprecated**
```

Comment:
This is listed as deprecated, but don't we still set the stop reason in a few places (i.e. https://github.com/livepeer/go-livepeer/blob/4589a1364fa9d29e9d196d259f1f235116d45953/eth/eventservices/jobservice.go#L133)?

Reply:
With the current (old) network protocol, the transcoder initiates the connection, so it needs heuristics to know when not to retry a job. With the new network, the transcoder doesn't really have the notion of stopping a job; it just receives segments and checks for the validity of each segment from there. As as result, more types of errors become recoverable, such as 'insufficient deposit'.

Perhaps we could continue using `stopReason` to short-cut certain checks by adding a "hard stop" for the entire job (eg , [inconsistent broadcasters](https://github.com/livepeer/go-livepeer/issues/371)), but I think that most of the existing error cases would go away with the new network.

Reply:
Removed the deprecated note, since it's not actually deprecated yet.